test/filt_stream.jl

```diff
@@ -275,6 +275,60 @@ function test_rational(h, x, ratio)
 end
 
 
+#
+# Integer resampling
+#
+
+function test_integer(h, x, ratio)
```

Comment:
It seems like the `test_xxx` functions are almost identical and could be combined into a single `test_resampling` function. Having a bunch of these almost-identical functions makes it hard to see what's actually different about the tests. This isn't strictly your problem because there's already quite a bit of duplication before this PR, but perhaps this is a good opportunity to try to refactor and clean up this testing. Would you be willing to take a crack at it?

I'd be happy to drop the big ASCII-art headings if that makes it easier to combine the functions. I don't know if anyone else is attached to them - maybe @simonster?

Reply:
It'd be very easy to combine the rational resampling and integer resampling (the rational resampling needs I think 1 change to run the integer tests). I had separated them because that seemed to be how the rest of the test functions were structured. I will edit this PR to reduce the additions. Refactoring all the tests is a bit beyond me. I'm trying to get my feet wet with my first PR here and I don't really understand all the testing machinery. I spent awhile looking at the tests and they have so much similarity there has to be was to reduce code duplication but I'm not sure how to go about it without obfuscating the process further or reducing code coverage. I also have a hard time prototyping a change as I'm not even sure how tests are run locally so I've been sort of manually testing the tests as it were.

Reply:
Yeah, that's reasonable. It's always a tough judgment call whether to duplicate for consistency, but the new version seems like a good balance - at least it's not making the duplication worse. :smile: 

I'll take a look tonight at the latest changes. I'll also see if I can figure out what's going on with all this `@timeifinteractive` stuff (it predates me working on this package). I usually just run the tests with `julia test/runtests.jl`.

Thanks for the contribution!